dask_kubernetes/operator/core.py

```diff
@@ -0,0 +1,167 @@
+import subprocess
+import json
+import re
+import time
+
+from distributed.core import rpc
+from distributed.deploy import Cluster
+
+from distributed.utils import Log, Logs
+
+from dask_kubernetes.auth import ClusterAuth
+from .daskcluster import (
+    build_cluster_spec,
+    build_worker_group_spec,
+)
+
+
+class KubeCluster2(Cluster):
+    """Launch a Dask Cluster on Kubernetes using the Operator"""
+
+    def __init__(
+        self,
+        name,
+        namespace="default",
+        image="daskdev/dask:latest",
+        replicas=3,
+        resources={},
+        env={},
+        asynchronous=False,
+        auth=ClusterAuth.DEFAULT,
+        **kwargs,
+    ):
+        self.name = name
+        self.namespace = namespace
+        self.core_api = None
+        self.custom_api = None
+        self.image = image
+        self.replicas = replicas
+        self.resources = resources
+        self.env = env
+        self.auth = auth
+
+        # TODO: Check if cluster already exists
+        super().__init__(asynchronous=asynchronous, **kwargs)
+        if not self.asynchronous:
+            self._loop_runner.start()
+            self.sync(self._start)
+
+    async def _start(self):
+        data = build_cluster_spec(
+            self.name, self.image, self.replicas, self.resources, self.env
+        )
+        with open("data.json", "w") as jfile:
```

Comment:
We should dump this to a [named temporary file](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile).